api/src/main/scala/org/ensime/api/incoming.scala

```diff
@@ -236,6 +236,16 @@ case class PrepareRefactorReq(
 ) extends RpcAnalyserRequest
 
 /**
+ * Responds with a `RefactorFailure` or a `RefactorDiffEffect`.
+ */
+case class DiffRefactorReq(
+  procId: Int,
+  tpe: Symbol, // tpe is ignored but part of the legacy wire format
+  params: RefactorDesc,
```

Comment:
ok, I see now that this gives us a really easy migration path onto diffs, which is the most important thing.